docs/data-sources/integration_custom_auth_action.md

```diff
@@ -0,0 +1,30 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "genesyscloud_integration_custom_auth_action Data Source - terraform-provider-genesyscloud"
+subcategory: ""
+description: |-
+  Data source for Genesys Cloud integration custom auth action. Select the custom auth action by its associated integration's name.
+---
+
+# genesyscloud_integration_custom_auth_action (Data Source)
+
+Data source for Genesys Cloud integration custom auth action. Select the custom auth action by its associated integration's name.
+
+## Example Usage
+
+```terraform
+data "genesyscloud_integration_custom_auth_action" "integrationCustomAuthAction" {
+  integration_name = "example integration name"
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `integration_name` (String) The name of the integration associated with the custom auth action
```

Comment:
changed it to `parent_integration_id` to lookup by `id` instead of `name`. 

To answer the question, this is used to lookup an existing `custom auth` by the integration it is attached to, then the custom auth action can be referenced in other parts of the terraform configuration. In terms of use case, there's really not much where you need to get an existing custom auth as `data source` for other resources, but we implement `data sources` with `resources` as general practice.